sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,173 @@
+	}
+
+	gem, err := &globalEndpointManager{
+		client:           client,
+		preferredRegions: preferredRegions,
+		locationCache:    newLocationCache(preferredRegions, *endpoint),
+	}, nil
+	if err != nil {
+		return nil, err
+	}
+
+	// Start the background refresh process with the specified refreshInterval.
+	gem.startBackgroundRefresh(refreshInterval)
+
+	return gem, nil
+}
+
+// GetWriteEndpoints returns the write endpoints from the location cache.
+func (gem *globalEndpointManager) GetWriteEndpoints() ([]url.URL, error) {
+	return gem.locationCache.writeEndpoints()
+}
+
+// GetReadEndpoints returns the read endpoints from the location cache.
+func (gem *globalEndpointManager) GetReadEndpoints() ([]url.URL, error) {
+	return gem.locationCache.readEndpoints()
+}
+
+// GetAccountProperties retrieves account properties from the Cosmos DB instance.
+func (gem *globalEndpointManager) GetAccountProperties() (accountProperties, error) {
+	operationContext := pipelineRequestOptions{
+		resourceType:    resourceTypeDatabaseAccount,
+		resourceAddress: "",
+	}
+
+	path, err := generatePathForNameBased(resourceTypeDatabaseAccount, "", false)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to generate path for name-based request: %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	azResponse, err := gem.client.sendGetRequest(path, ctx, operationContext, nil, nil)
+	cancel()
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to send GET request: %v", err)
+	}
+
+	properties, err := newAccountProperties(azResponse)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to parse account properties: %v", err)
+	}
+
+	return properties, nil
+}
+
+// newAccountProperties parses the account properties from the HTTP response.
+func newAccountProperties(azResponse *http.Response) (accountProperties, error) {
+	properties := accountProperties{}
+	err := azruntime.UnmarshalAsJSON(azResponse, &properties)
+	if err != nil {
+		return properties, err
+	}
+	return properties, nil
+}
+
+// GetLocation returns the location for the given endpoint from the location cache.
+func (gem *globalEndpointManager) GetLocation(endpoint url.URL) string {
+	return gem.locationCache.getLocation(endpoint)
+}
+
+// MarkEndpointUnavailableForRead marks an endpoint as unavailable for read operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForRead(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForRead(endpoint)
+}
+
+// MarkEndpointUnavailableForWrite marks an endpoint as unavailable for write operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForWrite(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForWrite(endpoint)
+}
+
+// Update updates the location cache and the client's default endpoint based on the provided account properties.
+func (gem *globalEndpointManager) Update() error {
```

Comment:
If we move forward with the Policy approach, this method needs to be guarded for concurrency. Because there could be multiple requests going through the Policy, detecting an old account information, and triggering an update concurrently.